src/field_message.c

```diff
@@ -0,0 +1,99 @@
+#include "field_message.h"
+
+#include <nitro.h>
+#include <string.h>
+
+#include "bg_window.h"
+#include "font.h"
+#include "game_options.h"
+#include "render_text.h"
+#include "render_window.h"
+#include "strbuf.h"
+#include "text.h"
+
+void FieldMessage_LoadTextPalettes(u32 palLocation, u32 resetPrinters)
+{
+    if (resetPrinters == TRUE) {
+        Text_ResetAllPrinters();
+    }
+
+    Font_LoadTextPalette(palLocation, 13 * 32, HEAP_ID_FIELD);
+    Font_LoadScreenIndicatorsPalette(palLocation, 12 * 32, HEAP_ID_FIELD);
+}
+
+void FieldMessage_AddWindow(BgConfig *bgConfig, Window *window, u32 bgLayer)
+{
+    if (bgLayer == BG_LAYER_MAIN_3) {
+        Window_Add(bgConfig, window, BG_LAYER_MAIN_3, 2, 19, 27, 4, 12, (((1024 - (18 + 12) - 9 - (32 * 8)) - (18 + 12 + 24)) - (27 * 4)));
+    } else {
+        Window_Add(bgConfig, window, BG_LAYER_SUB_3, 2, 19, 27, 4, 12, (512 - (27 * 4)));
+    }
+}
+
+void FieldMessage_DrawWindow(Window *window, const Options *options)
+{
+    LoadMessageBoxGraphics(window->bgConfig, Window_GetBgLayer(window), 1024 - (18 + 12), 10, Options_Frame(options), 4);
+    FieldMessaage_FillWindowTilemap(window);
+    Window_DrawMessageBoxWithScrollCursor(window, FALSE, 1024 - (18 + 12), 10);
+}
+
+void FieldMessaage_FillWindowTilemap(Window *window)
+{
+    Window_FillTilemap(window, 15);
+}
+
+u8 FieldMessage_Print(Window *window, Strbuf *strBuf, const Options *options, u8 canSkipDelay)
+{
+    RenderControlFlags_SetCanABSpeedUpPrint(canSkipDelay);
+    RenderControlFlags_SetAutoScrollFlags(FALSE);
+    RenderControlFlags_SetSpeedUpOnTouch(FALSE);
+
+    return Text_AddPrinterWithParams(window, FONT_MESSAGE, strBuf, 0, 0, Options_TextFrameDelay(options), NULL);
+}
+
+// used only in one instance
+u8 FieldMessage_PrintWithParams(Window *window, Strbuf *strBuf, int fontID, int renderDelay, u8 canSkipDelay, BOOL autoScroll)
+{
+    RenderControlFlags_SetCanABSpeedUpPrint(canSkipDelay);
+    RenderControlFlags_SetAutoScrollFlags(autoScroll);
+    RenderControlFlags_SetSpeedUpOnTouch(FALSE);
+
+    return Text_AddPrinterWithParams(window, fontID, strBuf, 0, 0, renderDelay, NULL);
+}
+
+u8 FieldMessage_FinishedPrinting(u8 printerID)
+{
+    if (Text_IsPrinterActive(printerID) == FALSE) {
+        return TRUE;
+    }
+
+    return FALSE;
```

Comment:
**question**: Does this match as `return Text_IsPrinterActive(printerID) != FALSE)`?

Reply:
from what i can tell, nope!